src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1415,10 +1426,20 @@ void Ekf2::Run()
 					aligned_ev_odom.y = aligned_pos(1);
 					aligned_ev_odom.z = aligned_pos(2);
 
-					const Vector3f aligned_vel = ev_rot_mat * Vector3f(_ev_odom.vx, _ev_odom.vy, _ev_odom.vz);
-					aligned_ev_odom.vx = aligned_vel(0);
-					aligned_ev_odom.vy = aligned_vel(1);
-					aligned_ev_odom.vz = aligned_vel(2);
+					if (_ev_odom.velocity_frame == _ev_odom.BODY_FRAME_FRD) {
+						Vector3f aligned_vel = Dcmf(_ekf.getQuaternion()) * Vector3f(_ev_odom.vx, _ev_odom.vy, _ev_odom.vz);
+						aligned_ev_odom.vx = aligned_vel(0);
+						aligned_ev_odom.vy = aligned_vel(1);
+						aligned_ev_odom.vz = aligned_vel(2);
+
+					} else { // _ev_odom.velocity_frame == _ev_odom.LOCAL_FRAME_FRD
```

Comment:
Just so we have less surprises at some point in the future

```suggestion
					} else if ( _ev_odom.velocity_frame == _ev_odom.LOCAL_FRAME_FRD) {
```